### PR TITLE
Rename library to ci-environment

### PR DESCRIPTION
**Pull request by aslakhellesoy**
This is a fix for #33. Summary of changes (I ended up doing some cleanup as well)

* No more dependency on `messages`
* Java: Use an interface for `CiEnvironment`
* Ruby: Use a `Hash` with symbol keys to represent `CiEnvironment`
* Rename `ciDict.json` to `CiEnvironments.json` and make it an array
* Rake: use a `file` task to copy `CiEnvironments.json` (and remove the github action task and rspec helper that did it)
* Update docs
* Rename package and module names
* Deleted some obsolete files

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
@aslakhellesoy I would suggest to "Use an array to define ci environments" in a distinctive PR as it also impacts ruby and java implementation.

That way, this PR, and a new one dedicated to using an array to define the CI environments would be easier to review

**Comment by aslakhellesoy** [comment]
I'm pretty deep into this - I think that would be too much to back out of now. We have good acceptance tests, so I'm not worried that switching to an array will break anything. This PR will also be a general cleanup.

File: .github/workflows/release-ruby.yml [hunk 1]
```diff
@@ -21,11 +21,6 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-
-      - name: Synchronize CI Dict
-        working-directory: ruby
-        run: bundle exec rake copy_ci_dict
-
```
Comment (aurelien-reeves) [inline .github/workflows/release-ruby.yml:28]: I am not sure that step can be removed without risk

**Review by aurelien-reeves — CHANGES_REQUESTED** [review]
The rake task to copy the json file should not be removed.
We can still want to execute rspec tests without using rake, so having it in the helper and required in the .rspec file would remain helpful
Also publishing the gem do not use rake, so executing the task just before publishing it is also required to be sure we release the same thing that what was tested before

File: .github/workflows/release-ruby.yml (hunk 1 above)
Comment (aslakhellesoy) [reply .github/workflows/release-ruby.yml:28]: Ok, fair point. I suggest:

- Restore `run: bundle exec rake lib/cucumber/ci_environment/CiEnvironments.json`
- Add a step at the end of the CI build that fails the build if there are dirty files in the repo.

Comment (aurelien-reeves) [reply .github/workflows/release-ruby.yml:28]: Agree

**Review by aurelien-reeves — APPROVED** [review]

**Pull request merged**